MSAL/src/native_auth/network/errors/sign_up/MSALNativeAuthSignUpChallengeResponseError.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+struct MSALNativeAuthSignUpChallengeResponseError: MSALNativeAuthResponseError {
+    let error: MSALNativeAuthSignUpChallengeOauth2ErrorCode
+    let errorDescription: String?
+    let errorCodes: [Int]?
+    let errorURI: String?
+    let innerErrors: [MSALNativeAuthInnerError]?
+
+    enum CodingKeys: String, CodingKey {
+        case error
+        case errorDescription = "error_description"
+        case errorCodes = "error_codes"
+        case errorURI = "error_uri"
+        case innerErrors = "inner_errors"
+    }
+}
+
+extension MSALNativeAuthSignUpChallengeResponseError {
+
+    func toSignUpPasswordStartPublicError() -> SignUpPasswordStartError {
+        switch error {
```

Comment:
Should there be a default case here in case the error returned by server is not considered here?

Reply:
Good question. The error handling is going to be refactored due to changes proposed by the API team, so we'll create a PBI in our backlog to add your suggestion in this code change. FYI @nilo-ms

Reply:
Will it impact what you return to the developers in the public API? If yes, this will be considered a breaking change per semantic versioning, and we'd want to avoid it.

Reply:
Hi @ameyapat , the method "toSignUpPasswordStartPublicError:" is used to convert the API error to the public SignUpPasswordStartError, so the field "error" can have only the values listed in the `MSALNativeAuthSignUpChallengeOauth2ErrorCode` enum. In case we receive a not expected error value from the server, the parse of the `MSALNativeAuthSignUpChallengeResponseError` would fail and we will return a "generalError" to the external developer.
@oldalton , the new eSTS error response structure is still under discussion, we will try our best to don't change the public interface. Anyway, we're internally investigating a way to make our SDK interface design more extendable, in order to avoid breaking changes